tests/consent/api/test_views.py

```diff
@@ -0,0 +1,452 @@
+            400
+        ),
+        # Missing `enterprise_customer` input.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': True,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': None,
+                'course_id': TEST_COURSE,
+            },
+            400
+        ),
+        # No consent in an enterprise course enrollment.
+        # Expect new consent resource to be created with consent provided.
+        (
+            factories.EnterpriseCustomerUserFactory,
+            [{
+                'user_id': TEST_USER_ID,
+                'enterprise_customer__uuid': TEST_UUID
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+            },
+            200
+        ),
+        # Consent already given for an enterprise course enrollment.
+        # Expect nothing to change.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': True,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+            },
+            200
+        ),
+        # Consent not given for an enterprise course enrollment.
+        # Expect consent to be given.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': False,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
```

Comment:
Let's return a value indicating the state as we would for a GET request with the given parameters.